libs/utils/src/sync/gate.rs

```diff
@@ -88,11 +106,20 @@ impl Gate {
     /// to avoid blocking close() indefinitely: typically types that contain a Gate will
     /// also contain a CancellationToken.
     pub fn enter(&self) -> Result<GateGuard, GateError> {
-        self.sem
-            .clone()
-            .try_acquire_owned()
-            .map(GateGuard)
-            .map_err(|_| GateError::GateClosed)
+        let permit = self
+            .inner
+            .sem
+            .try_acquire()
+            .map_err(|_| GateError::GateClosed)?;
+
+        // we now have the permit, let's disable the normal raii functionality and leave
+        // "returning" the permit to our GateGuard::drop
+        permit.forget();
+
+        Ok(GateGuard {
+            span_at_enter: tracing::Span::current(),
```

Comment:
How expensive is this tracing::Span::current()?  Not sure if it's copying a lot of stuff, or if this is a lightweight refcounted thing under the covers.

Reply:
My understanding is that for us memory-wise it's pointer sized. For the event which passes through to tracing-subscriber and what it needs to do to maintain the spans is probably expensive, but I don't think I've ever understood.

Found it, it's refcounted: https://docs.rs/tracing-subscriber/latest/src/tracing_subscriber/registry/sharded.rs.html#308-328